src/main/java/com/evokly/kafka/connect/mqtt/MqttSourceTask.java

```diff
@@ -1,34 +1,156 @@
-package com.evokly.kafka.connect.mqtt;
 /**
  * Copyright 2016 Evokly S.A.
- *
- * <p>See LICENSE file for License</p>
+ * See LICENSE file for License
  **/
 
+package com.evokly.kafka.connect.mqtt;
+
+import com.evokly.kafka.connect.mqtt.util.Version;
+
 import org.apache.kafka.connect.source.SourceRecord;
 import org.apache.kafka.connect.source.SourceTask;
 
+import org.eclipse.paho.client.mqttv3.IMqttDeliveryToken;
+import org.eclipse.paho.client.mqttv3.MqttCallback;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.eclipse.paho.client.mqttv3.persist.MemoryPersistence;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.Queue;
+
+/**
+ * MqttSourceTask is a Kafka Connect SourceTask implementation that reads
+ * from MQTT and generates Kafka Connect records.
+ */
+public class MqttSourceTask extends SourceTask implements MqttCallback {
+    private static final Logger log = LoggerFactory.getLogger(MqttSourceConnector.class);
+
+    MqttClient mClient;
+    String mMqttClientId = MqttClient.generateClientId();
+    Queue<MqttSourceTaskMessage> mQueue = new LinkedList<>();
 
-public class MqttSourceTask extends SourceTask {
+    /**
+     * Get the version of this task. Usually this should be the same as the corresponding
+     * {@link MqttSourceConnector} class's version.
+     *
+     * @return the version, formatted as a String
+     */
     @Override
     public String version() {
-        return null;
+        return Version.getVersion();
     }
 
+    /**
+     * Start the task.
+     *
+     * @param props initial configuration
+     */
     @Override
     public void start(Map<String, String> props) {
+        // Setup MQTT Client
+        MqttConnectOptions connectOptions = new MqttConnectOptions();
+
+        connectOptions.setCleanSession(true);
+        connectOptions.setKeepAliveInterval(30);
+        connectOptions.setServerURIs(props.get(MqttSourceConstant.MQTT_BROKER_URLS).split(","));
+
+        // Connect to Broker
+        try {
+            mClient = new MqttClient("tcp://127.0.0.1:1883", mMqttClientId,
```

Comment:
It is... setServerURIs override data from constructor. Can't set empty string because of validation.